### ISSUE TITLE
All Worms bug: worm name disappear

### ISSUE THREAD

**Issue by AdityaMH** [hrydgard/ppsspp#12380]
During the match (no matter what match or what worms game series), the name of the worm disappeared even though it changed option "show: name and health".

Android 7.1, OpenGL, version 9.1.2

**Comment by unknownbrackets** [comment]
Could you try exporting a GE frame dump?  These help a lot.

See here for instructions - it's not hard and works on Android too:
https://github.com/hrydgard/ppsspp/wiki/How-to-create-a-frame-dump

You can zip that and then drag and drop it into a reply here.

-[Unknown]

**Comment by TheThurbo** [comment]
Hey, so I can second this issue. It's a problem in the entire Worms series on PSP and has been a problem since I first used this emulator many years ago.

Worms have name tags on top of their health so you can differentiate them for the purposes of turn order. Curiously, all of them are either sometimes missing or display weird, corrupted-looking incorrect textures throughout the entire series. One game in the series, Worms Battle Islands, has an additional problem where the landscape pre-renderer doesn't work seemingly in a similar way as the names, making the "Tactics" mode unusable as you cannot preview the maps to spawn your gadgets.

Should I do a frame dump? The names are permanently missing or rendering badly at all times, I'm not sure how the frame dump would work/help. The issue is in plain sight the entire time.

Sorry if what I say is confusing or of little help.

**Comment by hrydgard** [comment]
Do make a frame dump, but it could be one of those issues where something happened in an early frame that ruined it for later frames so it's not always possible to tell.

**Comment by TheThurbo** [comment]
[Worms Battle Islands Frame Dumps.zip](https://github.com/hrydgard/ppsspp/files/5612425/Worms.Battle.Islands.Frame.Dumps.zip)

Did frame dumps for Battle Islands. I couldn't get the name tag textures to corrupt but it definitely happens in the other two games. Either way the names are missing and the landscape preview is also broken.

**Comment by LunaMoo** [comment]
One of those dumps just crashes for me.
 Either way the bug is real, the game tries to draw names from a texture stored in ram, but it's wrong and I can't see how it creates that texture, restarting battle doesn't re-generate the texture, but the address doesn't change when quitting the battle and starting new one so the function that writes it can be found:
![PPSSPP worms](https://user-images.githubusercontent.com/5485237/100551159-47b1c480-327f-11eb-8e2b-86b873a6d6dd.jpg)
I guess it should be downloading that texture from 04088000.

**Comment by unknownbrackets** [comment]
#13938 may at least fix the text.  Probably the landscape render is using another direct CPU copy which would have to be found.  It's also possible different Worms games will use variants of that func, since it was really inefficient...

-[Unknown]

**Comment by Panderner** [comment]
v1.11.2-129-gbcbfee640 Still happens
![Screenshot_2021-02-20-20-35-52-164_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/108595533-58279600-73bb-11eb-83c0-40a0b62ac0f0.jpg)

**Comment by unknownbrackets** [comment]
If you open the Disassembly (or web debugger), does the function named "worms_copy_normalize_alpha" exist?  You can just paste that name in the Go to field.

If it doesn't exist, it means that the non-demo version uses a slightly different function.

-[Unknown]

**Comment by Panderner** [comment]
I can't find worms_copy_normalize_alpha in Web debugger since I don't have a PC.

**Comment by Gamemulatorer** [comment]
This game is broken on vulkan backend graphics while on opengl backend looks good.

***Vulkan***
![Screenshot_2021-10-21-17-33-51-089_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/138252000-506291bf-974a-4442-85dc-31a7168a2d20.jpg)

***Opengl***
![Screenshot_2021-10-21-17-35-55-862_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/138252124-63f24aea-24b4-46a5-82c2-9f7c794545a2.jpg)

***Software***
![Screenshot_2021-10-21-17-33-31-045_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/138252200-be3b468c-b544-4a59-b251-28f20c6a587c.jpg)

**ge dump**
[worms_vulkan.zip](https://github.com/hrydgard/ppsspp/files/7387850/worms_vulkan.zip)

**Comment by hrydgard** [comment]
Cool. Let's close this then.

**Comment by Panderner** [comment]
How about worm name issue?
![NPEH00019_00000](https://user-images.githubusercontent.com/37503397/191250361-e0e7499b-ef31-4ebb-b786-bc08209d35c0.jpg)

**Comment by hrydgard** [comment]
Oh, OK, I thought both problems were fixed. reopening

**Comment by light-and-ray** [comment]
I have the problem with background on the last google play version of ppsspp

**Comment by light-and-ray** [comment]
"Worms open warfare 1" is OK, but "open warfare 2" has this problem

**Comment by hrydgard** [comment]
Oh, ok. What device are you using?

**Comment by light-and-ray** [comment]
> Oh, ok. What device are you using?

Android 11, retroid pocket 3+

**Comment by Gamemulatorer** [comment]
> > Oh, ok. What device are you using?
> 
> Android 11, retroid pocket 3+

That's a Mali gpu?
If yes try geometry culling

**Comment by light-and-ray** [comment]
> > > Oh, ok. What device are you using?
> > 
> > Android 11, retroid pocket 3+
> 
> That's a Mali gpu?
> If yes try geometry culling 

Yes, Mali

I got problem with semi transparent background on the winter map in the first game (after map changing). Geometry culling didn't help

In the second game geometry culling didn't help too

![IMG_20230219_121202.jpg](https://user-images.githubusercontent.com/33491867/219936755-a4030f82-3a0b-4657-ac29-2561783cc5c9.jpg)

![IMG_20230219_121216.jpg](https://user-images.githubusercontent.com/33491867/219936772-ba388a81-3a64-420e-a050-7c7bb6d949f0.jpg)

![IMG_20230219_121300.jpg](https://user-images.githubusercontent.com/33491867/219936777-2fede285-8c4e-4ffb-abed-ce36528898bd.jpg)

![IMG_20230219_121307.jpg](https://user-images.githubusercontent.com/33491867/219936781-a2630352-6c45-46cb-8f9e-596104f5164a.jpg)

**Comment by Gamemulatorer** [comment]
@light-and-ray how about if you try the recently build https://buildbot.orphis.net/ppsspp/

**Comment by light-and-ray** [comment]
> @light-and-ray how about if you try the recently build https://buildbot.orphis.net/ppsspp/
> There's a chance this might be fix by https://github.com/hrydgard/ppsspp/pull/16645

OK i will, but tomorrow

**Comment by light-and-ray** [comment]
> @light-and-ray how about if you try the recently build https://buildbot.orphis.net/ppsspp/ There's a chance this might be fix by #16645 nvm this is already included in 1.14.4 sweat_smile

Yes, it works. This commit megred "on Dec 23, 2022", and Google Play says last update was 3rd Jan. So I misunderstood

**Comment by light-and-ray** [comment]
P.s. on the last build it does not matter "Geometry shader culling" is checked or not

**Comment by Gamemulatorer** [comment]
> P.s. on the last build it does not matter "Geometry shader culling" is checked or not

Try this https://github.com/hrydgard/ppsspp/issues/12380#issuecomment-948461687

**Comment by light-and-ray** [comment]
@Gamemulatorer No, I meant the problem (of backgrounds) is gone. It is only needed to wait update of google play version

**Comment by light-and-ray** [comment]
**Worms Battle Islands**: there should be a map preview (1:30: https://www.youtube.com/watch?v=-JUf1WUezoM)
![Screenshot_20230221-171747](https://user-images.githubusercontent.com/33491867/220356952-f3e19d72-40e6-48d2-b5d0-e4c209ad102a.png)

both OpenGL and Vulkan, yesterday build
